### ISSUE TITLE
Help with building application extension with IntelliJ IDEA

### ISSUE THREAD

**Issue by rpelletierrefocus** [Zimbra-Community/zimbra-rocket#11]
Forgive me and I appreciate your patience. I have never worked with Java applications. I have installed IntelliJ Idea and opened the project in directory  zimbra-rocket\extension.

I made the changes I needed in file Rocket.java and then build the program. 

However extension.jar remains unchanged. 

I am using OpenJDK 14

I have spent some time googling this but I may need some appliction specific pointers.

Any assistance would be greatly appreciated regarding how to create a "release"

My fork is here:
https://github.com/rpelletierrefocus/zimbra-rocket

I plan to add another property called domain that will allow for the names to be simply the username, without the domain portion. It is still a work in progress, but my total inexperience in building java applications is getting in the way.

**Comment by ghost** [comment]
I have now added build.xml and makefile for use with ant and gnu make.

Just git clone the project, cd into the extension folder and run make or ant jar. You must have ant installed for it to build.

Please use a version of Java that is supported by Zimbra, to make sure you have that, compile as described in:

https://github.com/Zimbra/zm-extension-guide
https://github.com/Zimbra/zm-extension-guide/releases/download/0.0.3/README.pdf

And especially:
https://github.com/Zimbra/zm-extension-guide#setting-up-your-build-system

**Comment by rpelletierrefocus** [comment]
@barrydegraaff Thank you so much for the pointers. I have successfully added the functionality I was looking for. 
I added another parameter in config.properties  
```
domaininusername=false
```
Then I just added a couple of simple if then statements so that the previous functionaly (adding the domain to the username replacing @ can be retained if chosen.)

For anyone interested, here is my fork.  I doubt my solution is the most elegant as I have never done any coding with Java, but it works for me.

Not sure if you care to have a pull request for my final changes (I still have a little garbage code that would need to be cleaned up) But I will do so if you are interested
